src/gui/qgsbrowserdockwidget.cpp

```diff
@@ -157,6 +157,15 @@ void QgsBrowserDockWidget::showEvent( QShowEvent *e )
   QgsDockWidget::showEvent( e );
 }
 
+void QgsBrowserDockWidget::keyPressEvent( QKeyEvent *event )
+{
+  if ( event->key() == Qt::Key_F2 )
+  {
+    renameFavorite();
```

Comment:
Just calling `renameFavorite()` here as it checks the current item anyway and in future, we can overload this more if need based on the current item type.